tasks/componentUpdateTasks.ts

```diff
@@ -0,0 +1,186 @@
+                const newUrl = oldUrl.replace(
+                    /Microsoft\.VisualStudio\.Copilot\.Roslyn\.LanguageServer-[^/]+?\.zip/,
+                    `Microsoft.VisualStudio.Copilot.Roslyn.LanguageServer-${version}.zip`
+                );
+                if (oldUrl !== newUrl) {
+                    pkg.runtimeDependencies[i].url = newUrl;
+                    updated = true;
+                    console.log(`Updated RoslynCopilot url:\n  ${oldUrl}\n-> ${newUrl}`);
+                } else {
+                    console.log('RoslynCopilot url already up to date.');
+                }
+                break;
+            }
+        }
+    }
+
+    if (!updated) {
+        console.log('No changes required to package.json; aborting PR creation.');
+        return;
+    }
+
+    // Persist package.json
+    fs.writeFileSync('package.json', JSON.stringify(pkg, null, 2) + '\n', { encoding: 'utf8' });
+
+    // Prepare git
+    const safeVersion = version.replace(/[^A-Za-z0-9_.-]/g, '-');
+    const branch = `update/roslyn-copilot-${safeVersion}`;
+
+    // Make this optional so it can be tested locally by using dev's information. In real CI user name and email are always supplied.
+    if (parsedArgs.userName) {
+        await git(['config', '--local', 'user.name', parsedArgs.userName]);
+    }
+    if (parsedArgs.email) {
+        await git(['config', '--local', 'user.email', parsedArgs.email]);
+    }
+
+    await git(['checkout', '-b', branch]);
+    await git(['add', 'package.json']);
+    await git(['commit', '-m', `chore: update RoslynCopilot url to ${version}`]);
+
+    const pat = process.env['GitHubPAT'];
+    if (!pat) {
+        throw 'No GitHub PAT found.';
+    }
+
+    const remoteRepoAlias = 'targetRepo';
+    await git(
+        ['remote', 'add', remoteRepoAlias, `https://${parsedArgs.userName}:${pat}@github.com/dotnet/vscode-csharp.git`],
+        // Note: don't print PAT to console
+        false
+    );
+    await git(['fetch', remoteRepoAlias]);
+
+    const lsRemote = await git(['ls-remote', remoteRepoAlias, 'refs/head/' + branch]);
+    if (lsRemote.trim() !== '') {
+        // If the localization branch of this commit already exists, don't try to create another one.
+        console.log(`##vso[task.logissue type=error]${branch} already exists in dotnet/vscode-csharp. Skip pushing.`);
+    } else {
+        await git(['push', '-u', remoteRepoAlias]);
+    }
+
+    // Create PR via Octokit
+    try {
+        const octokit = new Octokit({ auth: pat });
+        const listPullRequest = await octokit.rest.pulls.list({
+            owner: 'dotnet',
+            repo: 'vscode-csharp',
+        });
+
+        if (listPullRequest.status != 200) {
+            throw `Failed get response from GitHub, http status code: ${listPullRequest.status}`;
+        }
+
+        const title = `Update RoslynCopilot url to ${version}`;
+        if (listPullRequest.data.some((pr) => pr.title === title)) {
```

Comment:
I think we should check this much earlier and avoid doing any work or pushing any branches.

Reply:
Agreed and changed, thank you.